exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/LateralJoinBatch.java

```diff
@@ -0,0 +1,861 @@
+        " of batches");
+      leftUpstream = OK;
+      return OK_NEW_SCHEMA;
+    }
+    return OK;
+  }
+
+  /**
+   * Finalizes the current output container with the records produced so far before sending it downstream
+   */
+  private void finalizeOutputContainer() {
+    VectorAccessibleUtilities.setValueCount(container, outputIndex);
+
+    // Set the record count in the container
+    container.setRecordCount(outputIndex);
+    container.buildSchema(BatchSchema.SelectionVectorMode.NONE);
+
+    batchMemoryManager.updateOutgoingStats(outputIndex);
+    logger.debug("Number of records emitted: " + outputIndex);
+
+    // Update the output index for next output batch to zero
+    outputIndex = 0;
+  }
+
+  /**
+   * Check if the schema changed between provided newSchema and oldSchema. It relies on
+   * {@link BatchSchema#isEquivalent(BatchSchema)}.
+   * @param newSchema - New Schema information
+   * @param oldSchema -  - New Schema information to compare with
+   *
+   * @return - true - if newSchema is not same as oldSchema
+   *         - false - if newSchema is same as oldSchema
+   */
+  private boolean isSchemaChanged(BatchSchema newSchema, BatchSchema oldSchema) {
+    return (newSchema == null || oldSchema == null) || !newSchema.isEquivalent(oldSchema);
+  }
+
+  /**
+   * Validate if the input schema is not null and doesn't contain any Selection Vector.
+   * @param schema - input schema to verify
+   * @return - true: valid input schema
+   *           false: invalid input schema
+   */
+  private boolean verifyInputSchema(BatchSchema schema) {
+
+    boolean isValid = true;
+    if (schema == null) {
+      logger.error("Null schema found for the incoming batch");
+      isValid = false;
+    } else {
+      final BatchSchema.SelectionVectorMode svMode = schema.getSelectionVectorMode();
+      if (svMode != BatchSchema.SelectionVectorMode.NONE) {
+        logger.error("Incoming batch schema found with selection vector which is not supported. SVMode: {}",
+          svMode.toString());
+        isValid = false;
+      }
+    }
+    return isValid;
+  }
+
+  /**
+   * Helps to create the outgoing container vectors based on known left and right batch schemas
+   * @throws SchemaChangeException
+   */
+  private void setupNewSchema() throws SchemaChangeException {
+
+    logger.debug(String.format("Setting up new schema based on incoming batch. New left schema: %s" +
+        " and New right schema: %s", left.getSchema(), right.getSchema()));
+
+    // Clear up the container
+    container.clear();
+    leftSchema = left.getSchema();
+    rightSchema = right.getSchema();
+
+    if (!verifyInputSchema(leftSchema)) {
+      throw new SchemaChangeException("Invalid Schema found for left incoming batch");
+    }
+
+    if (!verifyInputSchema(rightSchema)) {
+      throw new SchemaChangeException("Invalid Schema found for right incoming batch");
+    }
+
```

Comment:
In case of other joins planner takes care of renaming columns which has same name from left and right side of join. I will expect same thing to happen in Lateral Join as well rather than explicit check here.

Reply:
fair enough.